File: bms/go/ach/moov_io_ach_Addenda11_EntryDetailSequenceNumberField/moov_io_ach_Addenda11_EntryDetailSequenceNumberField.go
package test

import (
	"strconv"
	"strings"
)

var (
	moov_io_ach_stringZeros map[int]string = moov_io_ach_populateMap(94, "0")
)

const ()

func (addenda11 *moov_io_ach_Addenda11) EntryDetailSequenceNumberField() string {
	return addenda11.numericField(addenda11.EntryDetailSequenceNumber, 7)
}

func (c *moov_io_ach_converters) numericField(n int, max uint) string {
	s := strconv.FormatInt(int64(n), 10)
	if l := uint(len(s)); l > max {
		return s[l-max:]
	} else {
		m := int(max - l)
		pad, exists := moov_io_ach_stringZeros[m]
		if exists {
			return pad + s
		}
		// slow path
		return strings.Repeat("0", m) + s
	}
}

func moov_io_ach_populateMap(max int, zero string) map[int]string {
	out := make(map[int]string, max)
	for i := 0; i < max; i++ {
		out[i] = strings.Repeat(zero, i)
	}
	return out
}

type moov_io_ach_Addenda11 struct {
	// ID is a client defined string used as a reference to this record.

	// TypeCode Addenda11 types code '11'

	// Originator Name contains the originators name (your company name / name)

	// Originator Street Address Contains the originators street address (your company's address / your address)

	// EntryDetailSequenceNumber contains the ascending sequence number section of the Entry
	// Detail or Corporate Entry Detail Record's trace number This number is
	// the same as the last seven digits of the trace number of the related
	// Entry Detail Record or Corporate Entry Detail Record.
	EntryDetailSequenceNumber int `json:"entryDetailSequenceNumber"`
	// validator is composed for data validation

	// converters is composed for ACH to GoLang Converters
	moov_io_ach_converters
}

type moov_io_ach_converters struct{}

File: translations_only/fail/go/ach/moov_io_ach_Addenda11_EntryDetailSequenceNumberField/mistral/Hinted/moov_io_ach_Addenda11_EntryDetailSequenceNumberField.rs


use std::collections::HashMap;
use std::fmt;
use std::str;

const ZEROS: &str = "0";

struct MoovIoAchConverters;

impl MoovIoAchConverters {
    fn numeric_field(&self, n: i32, max: u32) -> String {
        let s = n.to_string();
        let l = s.len() as u32;
        if l > max {
            return s[(l - max) as usize..].to_string();
        } else {
            let m = max - l;
            let pad = MoovIoAchAddenda11::get_zeros(m as usize);
            return pad + &s;
        }
    }
}

struct MoovIoAchAddenda11 {
    entry_detail_sequence_number: i32,
    _converters: MoovIoAchConverters,
}

impl MoovIoAchAddenda11 {
    fn entry_detail_sequence_number_field(&self) -> String {
        self.numeric_field(self.entry_detail_sequence_number, 7)
    }

    fn numeric_field(&self, n: i32, max: u32) -> String {
        MoovIoAchConverters.numeric_field(n, max)
    }

    fn get_zeros(n: usize) -> String {
        let mut out = HashMap::new();
        for i in 0..n {
            out.insert(i, "0".repeat(i));
        }
        out[&(n - 1)].clone()
    }
}

impl fmt::Display for MoovIoAchAddenda11 {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(
            f,
            "ID: ?, TypeCode: Addenda11 types code '11', Originator Name: ?, Originator Street Address: ?, EntryDetailSequenceNumber: {},",
            self.entry_detail_sequence_number
        )
    }
}

fn main() {
    let addenda11 = MoovIoAchAddenda11 {
        entry_detail_sequence_number: 123,
        _converters: MoovIoAchConverters,
    };
    println!("{}", addenda11);
    println!(
        "EntryDetailSequenceNumberField: {}",
        addenda11.entry_detail_sequence_number_field()
    );
}

